include/reactor-uc/schedulers/dynamic/scheduler_federated.h

```diff
@@ -0,0 +1,18 @@
+#ifndef SCHEDULER_FEDERATED_H
+#define SCHEDULER_FEDERATED_H
+#include "reactor-uc/queues.h"
+#include "reactor-uc/scheduler.h"
+#include "reactor-uc/schedulers/dynamic/scheduler.h"
+
+typedef struct DynamicSchedulerFederated DynamicSchedulerFederated;
+typedef struct EnvironmentFederated EnvironmentFederated;
+
+struct DynamicSchedulerFederated {
```

Comment:
Why do we need a FederatedScheduler?

I think this will explode into our face at some point. Does this scheduler now only does the polling as an extra thing? 

if yes, I would suggest we add a new function to the Environment, like `env->bookkeeping` that based on if the environment is federated or non-federated does polling or not.

Reply:
Good point. We can put this function, `acquire_tag` into the Environment and provide an implementation based on whether we are federated or not